### PR TITLE
✨Source Amazon Seller Partner: multiple updates

### PR DESCRIPTION
**Pull request by askarpets**
## What
This branch contains updates needed for certification. For more details, please see the following PRs:

- https://github.com/airbytehq/airbyte/pull/32591
- https://github.com/airbytehq/airbyte/pull/32560
- https://github.com/airbytehq/airbyte/pull/32639
- https://github.com/airbytehq/airbyte/pull/32541
- https://github.com/airbytehq/airbyte/pull/32738

## 🚨 User Impact 🚨
No breaking changes

## Pre-merge Actions

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

**Review by lazebnyi — APPROVED** [review]
lgtm

File: airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/schemas/GET_VENDOR_TRAFFIC_REPORT.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "title": "Vendor Traffic Report",
+  "description": "This report shares data on the customer traffic to the detail pages of the vendor's items both at an aggregated level (across the vendor's entire catalog of items) and at a per-ASIN level. Data is available for different date range aggregation levels: DAY, WEEK, MONTH, QUARTER, YEAR. Requests can span multiple date range periThe Net Pure Product Margin report shares data with vendors on Amazon's profit margin selling the vendor's items both at an aggregated level (across the vendor's entire catalog of items) and at a per-ASIN level. Data is available at different date range aggregation levels: DAY, WEEK, MONTH, QUARTER, YEAR. Requests can span multiple date range periods.ods.",
```
Comment (katmarkham) [inline airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/schemas/GET_VENDOR_TRAFFIC_REPORT.json:3]: are these typos?:
-periThe
-periods.ods.

**Review by katmarkham — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/schemas/GET_VENDOR_TRAFFIC_REPORT.json (hunk 1 above)
Comment (askarpets) [reply airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/schemas/GET_VENDOR_TRAFFIC_REPORT.json]: My apologies - this has been fixed.

**Review by katmarkham — APPROVED** [review]

**Comment by lamb1333** [comment]
The connector now seems broken. I'm worried something in this update related to the "lastupdatedate__dbt_alter" data format is creating an issue. Here's the error I'm seeing: **column "lastupdatedate__dbt_alter" is of type timestamp with time zone but expression is of type text.** 
image

**Comment by lamb1333** [comment]
I did and it didn't work. Same error. Screenshots attached.
Screenshot 2023-12-08 at 12 14 20 PM

Screenshot 2023-12-08 at 12 14 28 PM

**Comment by askarpets** [comment]
@lamb1333 could you please clarify what is the source of columns `purchasedate__dbt_alter` and `lastupdatedate__dbt_alter`? I couldn't find them in the schemas for the `Orders` and `OrderItems` streams.

**Comment by lamb1333** [comment]
@askarpets it looks like in both of those cases, the columns `purchasedate__dbt_alter` and `lastupdatedate__dbt_alter` are being auto-generated by a DBT operation related to updating each of the columns `purchasedate` and `lastupdatedate`. 

In this pull request 32833, the schemas were changed so that these columns now have the format `date-time`. But maybe the auto-generated columns required to update are still text type. If I had to guess, this seems like the heart of the issue. 

Maybe we need to cast the auto-generated `purchasedate__dbt_alter` and `lastupdatedate__dbt_alter` columns to timestamp, or revert the `purchasedate` and `lastupdatedate` columns back to text, so that they match?
Screenshot 2023-12-13 at 9 16 53 AM

**Pull request merged**